lib/saluki-core/src/observability/metrics.rs

```diff
@@ -154,6 +155,8 @@ async fn flush_metrics(flush_interval: Duration) {
 
     let state = RECEIVER_STATE.get().expect("metrics receiver should be set");
 
+    let mut histogram_samples = Vec::<f64>::new();
```

Comment:
Just a small optimization to reuse the buffer for holding histogram samples instead of throwing it away each time.